### DISCUSSION TITLE
how to show peaks data on render & load audio file on button click?

### DISCUSSION THREAD

**Discussion by Smitt2767** [katspaugh/wavesurfer.js#3080] [Q&A] [answered]
I want to show peaks data from an external JSON file at the time of component rendering but only want to load an audio file when the user first clicks on the play button.

**Comment by katspaugh** [comment, marked as answer]
You can first load it with an empty URL and peaks, then load again with the actual URL.

```js
wavesurfer.load('', peaks)

onclick = {
  wavesurfer.load('https://...', peaks)
}
```
